pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/ParticipantExitRoute.java

```diff
@@ -1,115 +1,75 @@
 package org.broadinstitute.dsm.route;
 
-import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
-import java.util.Optional;
 
-import com.google.api.client.http.HttpStatusCodes;
-import com.google.gson.JsonObject;
-import com.google.gson.JsonParser;
-import lombok.NonNull;
+import com.google.gson.Gson;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.StringUtils;
-import org.broadinstitute.dsm.DSMServer;
-import org.broadinstitute.dsm.db.DDPInstance;
 import org.broadinstitute.dsm.db.KitDiscard;
-import org.broadinstitute.dsm.db.KitRequestShipping;
 import org.broadinstitute.dsm.db.ParticipantExit;
-import org.broadinstitute.dsm.db.dao.ddp.instance.DDPInstanceDao;
-import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
-import org.broadinstitute.dsm.exception.ParticipantNotExist;
+import org.broadinstitute.dsm.exception.AuthorizationException;
+import org.broadinstitute.dsm.exception.DSMBadRequestException;
+import org.broadinstitute.dsm.route.request.ParticipantExitRequest;
 import org.broadinstitute.dsm.security.RequestHandler;
-import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.service.participant.ParticipantExitService;
 import org.broadinstitute.dsm.statics.RequestParameter;
 import org.broadinstitute.dsm.statics.RoutePath;
-import org.broadinstitute.dsm.statics.UserErrorMessages;
-import org.broadinstitute.dsm.util.DDPRequestUtil;
 import org.broadinstitute.dsm.util.UserUtil;
-import org.broadinstitute.lddp.handlers.util.Result;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import spark.Request;
 import spark.Response;
 
+@Slf4j
 public class ParticipantExitRoute extends RequestHandler {
 
-    private static final Logger logger = LoggerFactory.getLogger(ParticipantExitRoute.class);
-    private static final String ROLE = "participant_exit";
+    private static final String PARTICIPANT_EXIT_ROLE = "participant_exit";
 
     @Override
-    public Object processRequest(Request request, Response response, String userId) throws Exception {
-        String requestBody = request.body();
-        String realm = request.params(RequestParameter.REALM);
-        if (StringUtils.isNotBlank(realm)) {
-            if (UserUtil.checkUserAccess(realm, userId, ROLE, null)) {
-                return ParticipantExit.getExitedParticipants(realm).values();
-            } else {
-                response.status(500);
-                logger.warn("User with id={} could not access realm={} with role={}", userId, realm, ROLE);
-                return new Result(500, UserErrorMessages.NO_RIGHTS);
-            }
+    public Object processRequest(Request request, Response response, String userId) {
+        String requestMethod = request.requestMethod();
```

Comment:
Just to make sure I get it: instead of switching behavior based on the existence of the `realm` param, now we switch based on get/post?

Reply:
Yes. The former code was relying on an implementation byproduct rather than testing for the request method directly.